developers/weaviate/starter-guides/managing-collections/collections-scaling-limits.mdx

```diff
@@ -0,0 +1,173 @@
+---
+title: Do you have too many collections?
+sidebar_label: Scaling limits with collections
+sidebar_position: 1
+image: og/docs/more-resources.jpg
+# tags: ['performance']
+---
+
+:::info
+Added in `v1.25.34`, `v1.26.18`, `1.27.15`, `1.28.9`, `1.29.2`
+:::
+
+:::caution You hit the collections count limit?
+If you see an error that the collection count limit has been reached, it means you **can't create any more collections** because the number of collections is affecting performance. This limit exists to control indexing, storage, and management complexity.<br /><br />
+**Instead of simply raising the limit, consider rethinking your architecture**. If you really need to change the limit, use the [`MAXIMUM_ALLOWED_COLLECTIONS_COUNT`](/developers/weaviate/config-refs/env-vars) environment variable.
+:::
+
+This guide offers an overview of the available architectural choices of using **multi-tenancy** or defining a **dedicated collection** for each subset of data.
+
+Consider a scenario where you are developing a SaaS platform for product recommendations, allowing users (merchants) to import and vectorize their product data. The users of this platform require the ability to efficiently perform vector searches to generate personalized recommendations for users on their respective e-commerce platforms.
+
+One option is creating a dedicated collection for each user to store their product data. However, as the number of users grows, so does the number of collections, potentially leading to performance bottlenecks and increased operational complexity. This leads us to an important question: **How many collections is too many?**
```

Comment:
In retrospect, I wonder if the framing of the page should be "Tenants vs multiple collections", rather than "How many collections is too many". 

My thought is - even if you only have a few (say, 10) collections, if they are parts of the same dataset with the same settings etc., tenants are the way to go. 

Also - the idea that there are "too many" collection is a Weaviate-specific implementation layer detail, and perhaps would also depend on a user's system spec. 

Would it make more sense to talk about: "What is the right mental framework for choosing whether to use a collection per dataset, or a tenant"

Reply:
Agreed, the real question isn't:
- How many collections is too many?

but rather :
- Should you use "Multi-tenancy" or "One collection per dataset"?

If we agree that the wording here is ok.